istioctl/cmd/auth_test.go

```diff
@@ -73,6 +86,31 @@ func TestAuthUpgrade(t *testing.T) {
 
 	for _, c := range testCases {
 		command := fmt.Sprintf("experimental auth upgrade -f %s --service %s", c.in, strings.Join(c.services, ","))
-		runCommandAndCheckGolden(c.name, command, c.golden, t)
+		runCommandAndCheckGoldenFile(c.name, command, c.golden, t)
+	}
+}
+
+func TestAuthValidator(t *testing.T) {
+	testCases := []struct {
+		name     string
+		in       []string
+		expected string
+	}{
+		{
+			name:     "good policy",
+			in:       []string{"testdata/auth/authz-policy.yaml"},
```

Comment:
Why do we put testdata in two different locations? one in under "istioctl/cmd/testdata/auth/", another under "istioctl/pkg/auth/testdata".

Reply:
We're just have different test cases. There are test cases at the other location and I didn't want to duplicate test files, since they are the same.

Reply:
I mean, shall we merge these tests to a single location?

Reply:
Tests at `istioctl/pkg/auth/validator_test.go` test functionality at package level. The ones at `auth_test.go` test the whole commands (which use the functions in those packages), i.e. `istioctl ...`. There is a `TestCoverage` that requires `auth` package to have test coverage above some threshold. @yangminzhu Do you have any input? I still think we should test at both places.

Reply:
Yeah, we should have both tests. But could you change line 106 to use testdata under the same directory? It's better not to do cross-directory reference like this, you can copy the contents of the file (optional: it would be better make some changes in the file so we don't test the same contents twice)

Reply:
I am not talking about tests. I was talking about the "testdata" folders, which contains some yaml files and expected Envoy config. Should we merge these two "testdata" folders under istioctl?

Reply:
I think testdata should follow where the test is written, otherwise we will have a lot of files in one place and it will be very hard to navigate through. I can create new files for the `auth_test.go` like @yangminzhu suggested. Wdyt?